packages/react/src/ReactCurrentBatchConfig.js

```diff
@@ -8,17 +8,20 @@
  */
 
 import type {Fiber} from 'react-reconciler/src/ReactInternalTypes';
+import type {Transition} from 'react-reconciler/src/ReactFiberTracingMarkerComponent.new';
 
 type BatchConfig = {
   transition: number,
   _updatedFibers?: Set<Fiber>,
+  transitionInfo: Transition | null,
```

Comment:
Let's combine this with the `transition` field. We don't use the number for anything currently, we just check if it's set with `currentBatchConfig.transition === 0`. We can update those to `currentBatchConfig.transition === null` instead.

Then in all those places where you temporarily unset both transition and transitionInfo, you only need to unset a single field.